build.gradle

```diff
@@ -318,6 +318,18 @@ subprojects {
     options.compilerArgs << "-Werror"
     options.release = minJavaVersion
     addParametersForTests(name, options)
+
+    // Todo: Do we need to separate some components to not require JDK 17? For example,
+    // streams had to be bumped becuase of the dependency on tools.
```

Comment:
Streams should still be built and tested with Java 11. I think this means we're blocked by https://issues.apache.org/jira/browse/KAFKA-17857

Also we are going to probably conflict with https://issues.apache.org/jira/browse/KAFKA-17811, and we can add connect and mirror to that list of modules to build with 17 as the minimum.

Reply:
I can just keep this PR open until we are unblocked and periodically update it by merging trunk into the PR to make sure I resolve any conflicts so not a huge deal.